### PR TITLE
Add empty state to images block

### LINKED ISSUE #39681: [Enhancement] Add an empty state illustration to the Images block
We can make images more visible and easier to understand by adding an illustration to its empty state. It'd only be shown when there are no images added.

Related: https://github.com/woocommerce/woocommerce/issues/39680

**Design**

AkVNGImLgSqCObTQ3idVn7-fi-2798_312465

image

**Acceptance criteria**

- [ ] When there are no attributes, we show an empty state illustration in the Attributes block.
- [ ] The illustration is a SVG file with no text strings.
- [ ] The illustration is not shown on mobile.

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

**Acceptance criteria**

- [ ] When there are no attributes, we show the empty state illustration in the Attributes block.
- [ ] The illustration is not shown on mobile.

Closes #39681.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. In the `General` tab, scroll down to the `Images` section.
4. The images place holder should be visible.
5. Add an image. The images place holder should disappear.
6. Verify the images place holder looks good on smaller screens and disappears on mobile.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by mattsherman — CHANGES_REQUESTED** [review]
Similar to my comments on @nathanss's recent #40956...

- The image has untranslatable text in it -- putting text in an image should be avoided
- The image has hardcoded colors and fonts, etc. in it, that may not match the editor if the editor's styles are updated (either by us or by an extension)

I think a more robust implementation would be to not use an image, but rather render this placeholder in React (this is what Nathan ended up doing).

Please note, that according to @jarekmorawski in his comments on the other PR, it is important that we keep text in the empty state (but, it needs to be translatable!).

**Comment by octaedro** [comment]
Thank you @mattsherman for your review,

> The image has untranslatable text in it -- putting text in an image should be avoided

Good eyes, I fixed it in the commit [2afd0c8](https://github.com/woocommerce/woocommerce/pull/41165/commits/2afd0c887cd8663292d9c213bbfbc7700feae0a6)

> The image has hardcoded colors and fonts, etc. in it, that may not match the editor if the editor's styles are updated (either by us or by an extension)

I'm confused about this comment because the images don't have texts. The texts under the images are being added [here](https://github.com/woocommerce/woocommerce/pull/41165/files#diff-1cb80dbb7f4737102fbf21e38dc5e141b6adffc44369732b5181deb168808916R48) (not directly in the image).

> I think a more robust implementation would be to not use an image but rather render this placeholder in React (this is what Nathan ended up doing).

I'm not sure about changing the image to a grey square since we are using them for different things. AFAIK we use the grey square when we load an element. Here we want to show the user they can put an image there. Let's confirm with @jarekmorawski.

**Comment by mattsherman** [comment]
> Thank you @mattsherman for your review,
> 
> > The image has untranslatable text in it -- putting text in an image should be avoided
> 
> Good eyes, I fixed it in the commit https://github.com/woocommerce/woocommerce/pull/41165/commits/2afd0c887cd8663292d9c213bbfbc7700feae0a6
> 
> > The image has hardcoded colors and fonts, etc. in it, that may not match the editor if the editor's styles are updated (either by us or by an extension)
> 
> I'm confused about this comment because the images don't have texts. The texts under the images are being added https://github.com/woocommerce/woocommerce/pull/41165/files#diff-1cb80dbb7f4737102fbf21e38dc5e141b6adffc44369732b5181deb168808916R48 (not directly in the image).

Sorry about that... looked quickly and assumed they were in the image!

> > I think a more robust implementation would be to not use an image but rather render this placeholder in React (this is what Nathan ended up doing).
> 
> I'm not sure about changing the image to a grey square since we are using them for different things. AFAIK we use the grey square when we load an element. Here we want to show the user they can put an image there. Let's confirm with @jarekmorawski.

Ignore this comment... we are all good.

Will re-review right now!

File: packages/js/product-editor/src/blocks/product-fields/images/edit.tsx [hunk 1]
```diff
@@ -193,6 +194,12 @@ export function Edit( {
 		}
 	}
 
+	const isImageGalleryVisible =
+		propertyValue !== null &&
+		propertyValue !== undefined &&
```
Comment (mattsherman) [inline packages/js/product-editor/src/blocks/product-fields/images/edit.tsx]: You don't need these checks. `Array.isArray()` will return `false` for either `null` or `undefined`.

**Review by mattsherman — APPROVED** [review]
Nice work!

Left one comment. If you feel like changing it before merging, please do. Pre-approving!

**Pull request merged**